agent/agent.go

```diff
@@ -341,43 +176,43 @@ func (a *Agent) handlePolicy(p *policystorage.Policy) {
 		return
 	}
 
-	var target targetpkg.Target
-	var apm apmpkg.APM
-	var strategy strategypkg.Strategy
+	var targetInst targetpkg.Target
+	var apmInst apmpkg.APM
+	var strategyInst strategypkg.Strategy
 
 	// dispense plugins
-	targetPlugin, err := a.targetManager.Dispense(p.Target.Name)
+	targetPlugin, err := a.pluginManager.Dispense(p.Target.Name, plugins.PluginTypeTarget)
 	if err != nil {
 		logger.Error("target plugin not initialized", "error", err, "plugin", p.Target.Name)
 		return
 	}
```

Comment:
Yes 100% in similar lines to how I expect we will handle policies which are not valid. Surfacing these errors in an easier to find way is something I am keen to tackle soon; otherwise as you know operators need to watch the logs to understand the health.